databricks-jobs/dbjobs.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python
+
+import base64
+import json
+import os
+from polling import TimeoutException, poll
+import requests
+
+DOMAIN = os.getenv('DATABRICKS_URL', 'dbc-caf9527b-e073.cloud.databricks.com')
+JOBS_BASE_URL = 'https://%s/api/2.0/jobs/' % (DOMAIN)
+RUNS_BASE_URL = 'https://%s/api/2.0/jobs/runs/' % (DOMAIN)
+JENKINS_USER = 'dataops+databrick_jenkins@mozilla.com'
+DATABRICKS_TOKEN = os.environ['DATABRICKS_TOKEN']
+
+
+def jobs_rpc(action, jsonstring):
+    """ A helper function to make the JOBS API POST request, request/response is encoded/decoded as JSON
+
+    Args:
+        action: The jobs rest api endpoint e.g. create, list, delete, get, reset, run-now
+        jsonstring: String representation of json payload to POST to the jobs REST api
+
+    Returns:
+        dict holding json response
+
+    """
+    response = requests.post(
+        JOBS_BASE_URL + action,
+        headers={"Authorization": "Basic " + base64.standard_b64encode("token:" + DATABRICKS_TOKEN)},
+        data=jsonstring
+    )
+    return response.json()
+
+
+def runs_rpc(action, jsonstring):
+    """ A helper function to make the job RUNS API GET request, request/response is encoded/decoded as JSON
+
+    Args:
+        action: The job runs rest api endpoint e.g. list, get, export, cancel, get-output, delete
+        jsonstring: String representation of json payload to POST to the job runs REST api
+
+    Returns:
+        dict holding json response
+
+    """
+    response = requests.get(
+        RUNS_BASE_URL + action,
+        headers={"Authorization": "Basic " + base64.standard_b64encode("token:" + DATABRICKS_TOKEN)},
+        data=jsonstring
+    )
+    return response.json()
+
+
+def reset_config(jobid, jsonfile):
+    """ Resets job config for jobid to contents of jsonfile
+
+    Args:
+        jobid: The id of job to modify
+        jsonfile: Path to the file containing json job definition
+
+    """
+    with open(jsonfile) as f:
+        job_settings_map = json.load(f)
+    data = {}
+    data['job_id'] = jobid
+    data['new_settings'] = job_settings_map
+    json_string = json.dumps(data)
+    print('Attempting to update jobid {} with settings {}'.format(jobid, json_string))
+    jobs_rpc('reset', json_string)
+
+
+def create_job(jsonfile):
+    """ Creates a new job
+
+    Args:
+        jsonfile: Path to the file containing json job definition
+
+    Returns:
+        jobid: The new job's id number
+
+    """
+    with open(jsonfile) as f:
+        job_settings_map = json.load(f)
+    job_settings_string = json.dumps(job_settings_map)
+    response_json = jobs_rpc('create', job_settings_string)
+    return response_json['job_id']
+
+
+def stop_job(job_id):
+    """ Stops all runs of a job, polls until stopped
+
+    Args:
+        job_id: The id of the job to stop all runs for
+
+    """
+    data = {}
```

Comment:
You could probably just construct the dict from the beginning since most of these values are already available:

```
data = {
  'active_only' : True
   'job_id': job_id,
   'limit': 0
}